src/LevelLoader.cpp

```diff
@@ -0,0 +1,298 @@
+
+         dynamicGameObj->initComponents();
+         world.addDynamicGameObject(dynamicGameObj);
+      }
+   }
+
+   return 0;
+}
+
+std::shared_ptr<GameObject> LevelLoader::createGameObject(json obj,
+   GameObjectType objType) {
+   InputComponent* inputComponent = getInputComponent(obj);
+   PhysicsComponent* physicsComponent = getPhysicsComponent(obj);
+   RenderComponent* renderComponent = getRenderComponent(obj);
+   ActionComponent* actionComponent = getActionComponent(obj);
+   
+   std::shared_ptr<GameObject> gameObj = std::make_shared<GameObject>(
+      objType,
+      glm::vec3(obj["pos"]["x"], obj["pos"]["y"], obj["pos"]["z"]),
+      glm::vec3(obj["dir"]["x"], obj["dir"]["y"], obj["dir"]["z"]),
+      static_cast<float>(obj["vel"]),
+      glm::vec3(obj["scale"]["x"], obj["scale"]["y"], obj["scale"]["z"]),
+      inputComponent,
+      physicsComponent,
+      renderComponent,
+      actionComponent,
+      obj["deliverable"]
+   );
+
+   return gameObj;
+}
+
+InputComponent* LevelLoader::getInputComponent(json obj) {
+   if (obj["input-component"] == nullptr) {
+      return nullptr;
+   }
+
+   std::string componentName = obj["input-component"]["name"];
+
+   if (componentName == "PlayerInputComponent") {
+      return new PlayerInputComponent();
+   }
+
+   return nullptr;
+}
+
+PhysicsComponent* LevelLoader::getPhysicsComponent(json obj) {
+   // No physics component included
+   if (obj["physics-component"] == nullptr) {
+      return nullptr;
+   }
+
+   // Invalid physics component syntax
+   if (obj["physics-component"]["name"] == nullptr) {
+      std::cerr << "Warning - Invalid physics component syntax" << std::endl;
+      return nullptr;
+   }
+
+   std::string componentName = obj["physics-component"]["name"];
+
+   if (componentName == "WallPhysicsComponent") {
+      return new WallPhysicsComponent();
+   } else if (componentName == "PlayerPhysicsComponent") {
+      return new PlayerPhysicsComponent();
+   }
+
+   return nullptr;
+}
+
+RenderComponent* LevelLoader::getRenderComponent(json obj) {
+   ShapeManager& shapeManager = ShapeManager::instance();
+   MaterialManager& materialManager = MaterialManager::instance();
+
+   // No render component included
+   if (obj["render-component"] == nullptr) {
+      return nullptr;
+   }
+
+   // Invalid render component syntax
+   if (obj["render-component"]["name"] == nullptr ||
+       obj["render-component"]["shape"] == nullptr ||
+       obj["render-component"]["shader"] == nullptr ||
+       obj["render-component"]["material"] == nullptr) {
+      std::cerr << "Warning - Invalid render component syntax" << std::endl;
+      return nullptr;
+   }
+
+   std::string componentName = obj["render-component"]["name"];
+   std::string shapeName = obj["render-component"]["shape"];
+   std::string shaderName = obj["render-component"]["shader"];
+   std::string materialName = obj["render-component"]["material"];
+
+   if (componentName == "WallRenderComponent") {
```

Comment:
I noticed we didn't really have a list of all the components used, so inorder to integrate well with JSON I had to determine which components to be using. This seemed to work well although another option would be a component manager that given a type of component returns you the appropriate `*Component` instance. Again this was a bit out of scope for me so we can refactor this later if you all don't like this.